### PR TITLE
Added "@ckeditor/ckeditor-cloud-services-core" package to this repo

### LINKED ISSUE #924: Add @ckeditor/ckeditor-cloud-services-core to this repo
`@ckeditor/ckeditor-cloud-services-core` depends on `@ckeditor/ckeditor5-utils` and e.g. `@ckeditor/ckeditor5-easy-image` depends on `@ckeditor/ckeditor-cloud-services-core`. So, it'd be good to release `@ckeditor/ckeditor-cloud-services-core` together with the rest of our repos. Which means that this package should be added to our development env.

Unfortunately, it does not contain the `ckeditor5-*` prefix so there are issues like https://github.com/cksource/mgit2/issues/72 or issues with `npm run test` if the directory name doesn't start with `packages/ckeditor5-*`.

### PR DESCRIPTION
**Pull request by pomek**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Internal: Added `@ckeditor/ckeditor-cloud-services-core` package to this repository. Closes #924.

---

### Additional information

Requires:

- https://github.com/cksource/mgit2/pull/98 - allows overriding directories,
- https://github.com/ckeditor/ckeditor-cloud-services-core/pull/27 - preparing the repository (scripts, readme, etc.)

### CODE REVIEW

File: mgit.json [hunk 1]
```diff
@@ -42,5 +46,8 @@
     "@ckeditor/ckeditor5-upload": "ckeditor/ckeditor5-upload",
     "@ckeditor/ckeditor5-utils": "ckeditor/ckeditor5-utils",
     "@ckeditor/ckeditor5-widget": "ckeditor/ckeditor5-widget"
+  },
+  "overrideDirectoryNames": {
+    "@ckeditor/ckeditor-cloud-services-core": "ckeditor5-cloud-services-core"
```
Comment (Reinmar) [inline mgit.json]: I don't lime the fact that we're renaming package which name starts with `ckeditor-` to `ckeditor5-`. In my original proposal in https://github.com/cksource/mgit2/issues/72 I proposed to have a different directory structure, where all the prefixes (`ckeditor5-*`) are stripped. So, according to my comment, this setting here should be:

```
 "overrideDirectoryNames": {
      "@ckeditor/ckeditor5-cloud-services-core": "cloud-services-core"
  }
```

So, the actual directory name should be `cloud-services-core`.

**Comment by Reinmar** [comment]
https://github.com/cksource/mgit2/issues/72#issuecomment-511462285

**Comment by Reinmar** [comment]
I asked the team for a feedback on this and they see this a bit differently than I did:

image

So, we need to change our approach here a bit, unfortunately.

**Comment by Reinmar** [comment]
To adjust now: 

* [ ] Docs (API + guides)
* [ ] Manual tests and automatic tests
* [ ] Release
* [ ] CI

Followups: 

* [ ] Content styles
* [ ] Theme

**Comment by pomek** [comment]
**Docs**

I fixed as much as I could but I have no idea what to do with the errors below:

```
JSDoc started...
Incorrect param type: Blob
        at /Users/pomek/Projects/ckeditor/ckeditor5/packages/ckeditor-cloud-services-core/src/uploadgateway/fileuploader.js (line 29)

Incorrect param type: Blob
        at /Users/pomek/Projects/ckeditor/ckeditor5/packages/ckeditor-cloud-services-core/src/uploadgateway/uploadgateway.js (line 74)

Incorrect link: Token
        at /Users/pomek/Projects/ckeditor/ckeditor5/packages/ckeditor-cloud-services-core/src/token/token.js (line 150)

Invalid return type: Blob.                                                                                                                                                                                                                            at /Users/pomek/Projects/ckeditor/ckeditor5/packages/ckeditor-cloud-services-core/src/uploadgateway/fileuploader.js (line 248)


Validation Summary:
 * Validator found 4 errors.
```

Good news, https://github.com/ckeditor/ckeditor5/issues/1978 will be resolved after adding the package to our repos.

**Comment by pomek** [comment]
**Release tools**

image

image

**Comment by pomek** [comment]
**Content styles** works. I don't remember what I should check in the "Theme" topic but if anything won't work, MT and Content styles scripts would print any error.

**Comment by pomek** [comment]
After merging PRs: https://github.com/ckeditor/ckeditor5-dev/pull/554 and https://github.com/ckeditor/ckeditor-cloud-services-core/pull/27, CI should work.

**Comment by pomek** [comment]
Error codes from the package are visible in the error table:

image

**Comment by pomek** [comment]
I found this:

image

**Comment by pomek** [comment]
Fixed.

image

File: mrgit.json [hunk 2]
```diff
@@ -1,8 +1,8 @@
 {
   "packages": "packages/",
-  "packagesPrefix": "@ckeditor/ckeditor5-",
-  "baseBranches": [ "master", "stable" ],
```
Comment (Reinmar) [inline mrgit.json]: I guess you didn't want to remove this?

File: package.json [hunk 3]
```diff
@@ -21,6 +21,7 @@
     "framework"
   ],
   "dependencies": {
+    "@ckeditor/ckeditor-cloud-services-core": "^4.0.0",
```
Comment (Reinmar) [inline package.json]:
```suggestion
    "@ckeditor/ckeditor-cloud-services-core": "^5.0.0",
```

File: mrgit.json (hunk 2 above)
Comment (pomek) [reply mrgit.json]: Sure, I will fix that.

**Comment by Reinmar** [comment]
```
(t/924 b7bb733 M1) p@m /workspace/ckeditor5> yarn test --files=cloud-services-core -c
yarn run v1.17.3
$ node --max_old_space_size=4096 node_modules/@ckeditor/ckeditor5-dev-tests/bin/test.js --files=cloud-services-core -c
TypeError: glob.match is not a function
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
(t/924 b7bb733 M1) p@m /workspace/ckeditor5>
```

**Comment by pomek** [comment]
@Reinmar, the error has been fixed (https://github.com/ckeditor/ckeditor5-dev/commit/18feef7).

File: mrgit.json [hunk 4]
```diff
@@ -1,8 +1,9 @@
 {
   "packages": "packages/",
-  "packagesPrefix": "@ckeditor/ckeditor5-",
+  "packagesPrefix": [ "@ckeditor/ckeditor5-", "@ckeditor/ckeditor-" ],
   "baseBranches": [ "master", "stable" ],
   "dependencies": {
+    "@ckeditor/ckeditor-cloud-services-core": "ckeditor/ckeditor-cloud-services-core#t/ckeditor5/924",
```
Comment (Reinmar) [inline mrgit.json]:
```suggestion
    "@ckeditor/ckeditor-cloud-services-core": "ckeditor/ckeditor-cloud-services-core",
```

**Pull request merged**